### PR TITLE
Update satellite.py

### PR DESCRIPTION
**Pull request by rsriya**
Updated the satellite.py according to QA discussions of 1st May.

### CODE REVIEW

File: qnv.py [hunk 1]
```diff
@@ -1,37 +1,24 @@
...
-	v5 = np.dot(v3,v4)
-		
+def dot(v1,v2): #returns dot product of 2 row vectors
+
+	v5 = np.dot(v1,v2)		
 	return v5
 
-def cross1(v1,v2):
-	#for cross product of 2 column vectors
+def cross(v1,v2): #returns cross product of 2 row vectors
```
Comment (sanket789) [inline qnv.py]: Why is this needed?

File: qnv.py [hunk 2]
```diff
@@ -41,42 +28,47 @@ def quatMultiply(q1,q2):
...
 	b = a1*b2 + a2*b1 + np.cross(b1,b2)
-
-	#b = b.reshape((1,3))
-
 	q = np.hstack((a,b))
 	q = q/np.linalg.norm(q)
 	return q
 
-def quatRotate(q,x):
+def quatMultiplyunnorm(q1,q2): #returns quaternion product (product is not a unit quaternion)
```
Comment (sanket789) [inline qnv.py]: Make the name quatMultiplyUnnorm for better readability

File: qnv.py [hunk 3]
```diff
@@ -41,42 +28,47 @@ def quatMultiply(q1,q2):
...
 	y = np.hstack(([0.],x.copy()))
-	y = quatMultiply(q,y)
-	y = quatMultiply(y,qi)
-
+	y = quatMultiplyunnorm(q,y)
+	y = quatMultiplyunnorm(y,qi)
 	x2 = y[1:4]
 	return x2
 
 def quatDer1(q,w): #if w is in body frame, q takes from body to inertial
```
Comment (sanket789) [inline qnv.py]: change the comment to : w i angular velocity of body wrt inertial frame in body frame. q transforms inertial frame vector to body frame

File: qnv.py [hunk 4]
```diff
@@ -41,42 +28,47 @@ def quatMultiply(q1,q2):
...
 def quatDer2(q,w): #if w is in inertial frame, q takes from body to inertial
-	W = np.array([[0,-w[0],-w[1],-w[2]],[w[0],0,-w[2],w[1]],[w[1],w[2],0,-w[0]],[w[2],-w[1],w[1],0]])
+	W = np.array([[0,-w[0],-w[1],-w[2]],[w[0],0,-w[2],w[1]],[w[1],w[2],0,-w[0]],[w[2],-w[1],w[0],0]])
 	q_dot = 0.5*np.dot(W,q)
 
 	return q_dot
-
-def rotm2quat(A):
+'''
+def rotm2quat(A): #returns a quaternion whose scalar part is positive to keep angle between -180 to +180 deg.
```
Comment (sanket789) [inline qnv.py]: Can the quaternion notation be changed to [q0,q1,q2,q3] which matches with next function as well as with book

File: satellite.py [hunk 5]
```diff
@@ -7,20 +7,18 @@ class Satellite:
...
-		#print self.state ,"state_set"
 
 	def getState(self):	#returns the state
 
 		return self.state
 
-	def setPos(self,pos):	#set position in eci
+	def setPos(self,pos):	#set position in eci (earth centered inertial frame)
 
 		self.v_pos_i = pos
```
Comment (sanket789) [inline satellite.py]: pos.copy()

File: satellite.py [hunk 6]
```diff
@@ -119,15 +112,11 @@ def getOmega_m(self):
...
-
-
-
-'''
 	def getW(self):
 
-		return self.state[10:13]
+		return self.state[4:7]
 
+'''
```
Comment (sanket789) [inline satellite.py]: These three functions can be deleted (even from comments)

File: test_code_for_qnv.py [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+if G != 0:
+	flag=0
+
+if flag == 1:
+	print ("all cases passed for dot")
+else:
+	print ("error for dot")
+
+flag=1 #to check for cross
+cross=np.array([0,0,0])
```
Comment (sanket789) [inline test_code_for_qnv.py]: It's better if you don't have the name for variable same as the function. change it to `expected_cross` etc

File: qnv.py (hunk 1 above)
Comment (rsriya) [reply qnv.py]: For quatMultiply we need to do cross product of 2 vectors.

**Comment by sanket789** [comment]
Merged the changes from your pull request.
